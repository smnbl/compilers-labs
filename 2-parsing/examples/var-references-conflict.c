int varrefsconflict()
{
    int var = 4;
    int arr[4];

    var;
    arr[0];
}
