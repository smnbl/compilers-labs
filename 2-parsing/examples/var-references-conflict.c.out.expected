└── Program
    └── FuncDecl: returnType = 'int', name = 'varrefsconflict'
        └── CompoundStmt
            ├── VarDecl: type = 'int', name = 'var'
            │   └── IntLiteral: value = '4'
            ├── ArrayDecl: type = 'int', name = 'arr'
            │   └── IntLiteral: value = '4'
            ├── ExprStmt
            │   └── VarRefExpr: name = 'var'
            └── ExprStmt
                └── ArrayRefExpr: name = 'arr'
                    └── IntLiteral: value = '0'
